vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
OK - I think none of the changes actually followed the recommendation above

```
a = calloc(nrecs, sizeof(*(a)));
b = malloc(nrecs * sizeof(*(b)));
```

Even though it results in rather ugly code (as discussed above), it would be much easier to just apply this pattern consistently to avoid ambiguity. i.e.

```
(*veg_hist)[i][j].LAI = calloc(NR + 1, sizeof(*((*veg_hist)[i][j].LAI))); // ugly, but certainly correct
```

rather than 
```
(*veg_hist)[i][j].LAI = calloc(NR + 1, sizeof(*(*veg_hist)[i][j].LAI));
```
which I think is more ambiguous (who remember whether `*` binds tighter than `.LAI` ...